### PR TITLE
Seller Experience: Customize Woo landing page for Sell users

### LINKED ISSUE #61472: Seller Experience - Customize Woo landing page for Sell users
For "Sell" users that have a launched site that is using a product block (i.e. they took the "Start Simple" path during onboarding), we should display customized content on the Woo landing page - CTA text “Upgrade your store” with a sub message “Need more out of your store? Unlock the tools needed to manage products, orders, shipping and more.”

image

### PR DESCRIPTION
**Pull request by markbiek**
#### Changes proposed in this Pull Request

Display custom content for simple sellers on the WooCommerce admin landing page. A simple seller is something who has a payment-block website and hasn't upgraded yet.

#### Testing instructions

Visit a non-seller website admin and go to `/woocommerce-installation`. You should see the following content:

image

Then visit the `/woocommerce-installation` page for a website that used the "Start simple" seller signup flow. You should see the following content instead:
image

Fixes #61472

### CODE REVIEW

File: client/state/sites/hooks/use-is-simple-seller.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useSelector } from 'react-redux';
+import { isSimpleSeller } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
```
Comment (sixhours) [inline client/state/sites/hooks/use-is-simple-seller.ts:4]: I got the impression from Roz's comment in Slack (p1646669765868609/1646668487.357219-slack-C0Q664T29) that we don't need a new selector for this, we can leverage `useSiteOption` directly...but maybe I misunderstood!

If you do go this route, I'd add a doc block here for clarity; it looks like we do with the other `sites` hooks.

File: client/state/sites/hooks/test/use-is-simple-seller.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import { renderHook } from '@testing-library/react-hooks';
```
Comment (sixhours) [inline client/state/sites/hooks/test/use-is-simple-seller.js:1]: Yay, tests! I ran these with `yarn run test-client client/state/sites/hooks/test/use-is-simple-seller.js` and everything passed.

File: client/state/sites/selectors/is-simple-seller.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * Returns true if the current site is a simple seller site (payment block, no Woo plan)
+ *
+ * @param {object} state Global state tree
+ * @returns {boolean}
+ */
+export default function isSimpleSeller( state: AppState, siteId: number ): boolean | null {
+	// Check if the site intent is 'sell'.
+	const intent = getSiteOption( state, siteId, 'site_intent' );
+	if ( ! intent ) {
+		return null;
```
Comment (blackjackkent) [inline client/state/sites/selectors/is-simple-seller.ts]: I think this should return false as well, for consistency.

File: client/my-sites/woocommerce/landing-page.tsx [hunk 4]
```diff
@@ -26,10 +27,18 @@ interface Props {
...
+	illustration: string;
+	line: string;
+	action: string;
+}
+
 const LandingPage: React.FunctionComponent< Props > = ( { siteId } ) => {
 	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce' } ];
 	const ctaRef = useRef( null );
+	const simple_seller = useIsSimpleSeller();
```
Comment (blackjackkent) [inline client/my-sites/woocommerce/landing-page.tsx]: Nitpick, but let's have this as `simpleSeller` to fit the Javascript camel-case convention.

**Review by sixhours — APPROVED** [review]
Minor note on the selector, but this is working well for me!

**Review by blackjackkent — APPROVED** [review]
A few small tweak suggestions but otherwise looks good! :D Nice work on adding tests as well.

File: client/state/sites/hooks/use-is-simple-seller.ts (hunk 1 above)
Comment (blackjackkent) [reply client/state/sites/hooks/use-is-simple-seller.ts:4]: You can use `useSiteOption` directly to get the intent. It looks like what's being done here is to combine the result of the site option (the intent) and also information about the user's plan, which requires its own hook. So I think this is okay.

Comment (markbiek) [reply client/state/sites/hooks/use-is-simple-seller.ts:4]: Yup, my intention was exactly what @blackjackkent is describing.

Good catch on the doc block though. I've added that in.

File: client/state/sites/selectors/is-simple-seller.ts (hunk 3 above)
Comment (markbiek) [reply client/state/sites/selectors/is-simple-seller.ts]: I was conflicted about that but I think you're right. Fixed!

File: client/my-sites/woocommerce/landing-page.tsx (hunk 4 above)
Comment (markbiek) [reply client/my-sites/woocommerce/landing-page.tsx]: Fixed!

File: client/state/sites/hooks/test/use-is-simple-seller.js (hunk 2 above)
Comment (markbiek) [reply client/state/sites/hooks/test/use-is-simple-seller.js:1]: In retrospect, I should've put a note in the testing instructions for the PR. Next time!

File: client/state/sites/hooks/use-is-simple-seller.ts (hunk 1 above)
Comment (andres-blanco) [reply client/state/sites/hooks/use-is-simple-seller.ts:4]: Example of `useSiteOption` usage: https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/inline-help/inline-help-search-results.jsx#L61

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7172239

Thank you @markbiek for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7172239).

**Pull request merged**